### PR TITLE
Move reports to its own controler

### PR DESCRIPTION
**Pull request by differentreality**
This is a follow-up for https://github.com/openSUSE/osem/pull/1169
And it includes https://github.com/openSUSE/osem/issues/1146

* Moves reports to its own controller
* Makes all links of reports in the admin namespace
* Adds ID column for events in report pages 
* Creates partial for every tab in reports#index
* Shows commercial tab in admin/events#show

### CODE REVIEW

**Comment by differentreality** [comment]
Any chance anyone wants to review those additions? :)

**Comment by differentreality** [comment]
Ping

File: app/views/admin/events/show.html.haml [hunk 1]
```diff
@@ -62,23 +64,25 @@
             %td{'class'=>class_for_todo(progress_status['registered'])}
               %span{'class'=>[icon_for_todo(progress_status['registered']), 'fa-lg']}
           %tr
-            %td= link_to 'Fill out submitter biography', edit_user_path(@event.submitter)
+            %td= link_to 'Fill out submitter biography', edit_admin_user_path(@event.submitter)
```
Comment (ChrisBr) [inline app/views/admin/events/show.html.haml]: How is this related to this PR?

File: app/views/admin/events/show.html.haml [hunk 2]
```diff
@@ -62,23 +64,25 @@
...
             %td{'class'=>class_for_todo(progress_status['biography'])}
               %span{'class'=>[icon_for_todo(progress_status['biography']), 'fa-lg']}
           %tr
-            %td= link_to 'Add a subtitle', edit_conference_program_proposal_path(@event.program.conference.short_title, @event)
+            %td= link_to 'Add a subtitle', edit_admin_conference_program_event_path(@event.program.conference.short_title, @event)
```
Comment (ChrisBr) [inline app/views/admin/events/show.html.haml]: Same here?

File: app/views/admin/events/show.html.haml [hunk 3]
```diff
@@ -62,23 +64,25 @@
...
             %td{'class'=>class_for_todo(progress_status['subtitle'])}
               %span{'class'=>[icon_for_todo(progress_status['subtitle']), 'fa-lg']}
           %tr
-            %td= link_to 'Add a commercial', edit_conference_program_proposal_path(@event.program.conference.short_title, @event, anchor: 'commercials-content')
+            %td= link_to 'Add a commercial', edit_admin_conference_program_event_path(@event.program.conference.short_title, @event, anchor: 'commercials-content')
```
Comment (ChrisBr) [inline app/views/admin/events/show.html.haml]: Same here?

File: app/views/admin/events/show.html.haml [hunk 4]
```diff
@@ -62,23 +64,25 @@
...
             %td{'class'=>class_for_todo(progress_status['commercials'])}
               %span{'class'=>[icon_for_todo(progress_status['commercials']), 'fa-lg']}
           - unless progress_status['track'].nil?
             %tr
-              %td= link_to 'Add a track', edit_conference_program_proposal_path(@event.program.conference.short_title, @event)
+              %td= link_to 'Add a track', edit_admin_conference_program_event_path(@event.program.conference.short_title, @event)
```
Comment (ChrisBr) [inline app/views/admin/events/show.html.haml]: Same here?

File: app/views/admin/events/show.html.haml [hunk 5]
```diff
@@ -62,23 +64,25 @@
...
               %td{'class'=>class_for_todo(progress_status['track'])}
                 %span{'class'=>[icon_for_todo(progress_status['track']), 'fa-lg']}
           %tr
-            %td= link_to 'Add a difficulty level', edit_conference_program_proposal_path(@event.program.conference.short_title, @event)
+            %td= link_to 'Add a difficulty level', edit_admin_conference_program_event_path(@event.program.conference.short_title, @event)
```
Comment (ChrisBr) [inline app/views/admin/events/show.html.haml]: Same here

**Comment by differentreality** [comment]
https://github.com/openSUSE/osem/pull/1322

**Comment by ChrisBr** [comment]
@differentreality wondering why the events views are not deleted?

**Comment by differentreality** [comment]
@ChrisBr good question :) Removed admin/events/reports.html.haml

**Comment by ChrisBr** [comment]
@differentreality factory linting fails? Any idea? Maybe rebase?

**Comment by ChrisBr** [comment]
Otherwise LGTM

**Comment by differentreality** [comment]
@ChrisBr ?

**Comment by differentreality** [comment]
@ChrisBr are you OK with the changes? Can we merge this?
